### PR TITLE
Check length of canonical handle in addition to the display handle

### LINKED ISSUE #1673: User handles doesn't strip leading/trailing whitespace
While leading and trailing whitespace are allowed (they are stripped before canonicalization), the length requirement is being broken by only checking before canonicalization instead of after.

image
image

### PR DESCRIPTION
**Pull request by wilwade**
# Goal
The goal of this PR is to require that handles be a minimum length after canonicalization.

Closes #1673

# Discussion
- Adds a test around whitespaces stripping to be more clear
- Renames the constants in the handles primitives

### CODE REVIEW

**Review by rlaferla — COMMENTED** [review]
Not related to this bug but do we want to add the delimiter to the `HandleResponse` in this PR?  @JoeCap08055

File: pallets/handles/src/lib.rs [hunk 1]
```diff
@@ -693,6 +692,17 @@ pub mod pallet {
 			Ok(())
 		}
 
+		fn validate_canonical_handle_length(canonical_handle_str: &str) -> DispatchResult {
```
Comment (enddynayn) [inline pallets/handles/src/lib.rs:695]: nice

File: pallets/handles/src/handles-utils/src/tests/converter_tests.rs [hunk 2]
```diff
@@ -111,6 +111,18 @@ fn test_convert_to_canonical_combining_marks_reduce_to_the_same_canonical_form()
 	assert_eq!(canonical_with_combining_mark, expected_canonical_form);
 }
 
+#[test]
+fn test_convert_to_canonical_handles_whitespace() {
+	let all_spaces = String::from("         ");
```
Comment (rlaferla) [inline pallets/handles/src/handles-utils/src/tests/converter_tests.rs:116]: Do we want to test other whitespace than just space?

**Review by rlaferla — APPROVED** [review]

**Review by enddynayn — APPROVED** [review]
👍 great!

**Review by shannonwells — APPROVED** [review]
Thanks, looks good.

**Review by saraswatpuneet — APPROVED** [review]
Awesome thank you

**Pull request merged**